ccf_transaction_processor/Readme.md

```diff
@@ -0,0 +1,46 @@
+<!---
+Licensed under Creative Commons Attribution 4.0 International License
+https://creativecommons.org/licenses/by/4.0/
+--->
+
+## Microsoft CCF based PDO Transaction Processor
+
+This folder contains software for PDO transaction processor (TP) based on Microsoft's CCF blockchain.
+The software is located under transaction_processor/. The folder CCF/ points to CCF tag 0.7.1 which is
+included as a submodule under the PDO repo. The TP software is written and tested for CCF tag 0.7.1.
+Compatability with other CCF versions is not guaranteed.
+
+The TP must viewed as a CCF application. Documentation for building and deploying CCF applications
+can be found at https://microsoft.github.io/CCF/. The CCF legder that stores the PDO TP registries is
+encrypted, and is accessible only within CCF encalves. Currently PDO/CCF combination is supported
+only under the virtual enclave mode for both PDO and CCF. (set env variable SGX_MODE=SIM for PDO &
+set cmake flag TARGET=virtual for ccf). Support for HW mode for both PDO and CCF will be added soon.
+
+CCF uses mutually authenticated TLS channels for transactions. Given that in PDO client authentication is implemented within the transaction processor itself, we do not utilize the client authentication feature provided by CCF. Once CCF is deployed, CCF's network certificate (networkcert.pem) and one set of user keys (userccf_cert.pem & userccf_privk.pem)
+must be made available to all PDO processes that want to submit a CCF transaction. In this case, every
+PDO process behaves as though it is a CCF user corresponding to the private key userccf_privk.pem.
+These keys must be stored under $PDO_LEDGER_KEY_ROOT as part of PDO deployment.
+
+It may be noted that PDO also supports TP based on the Hyperledger Sawtooth blockchain.
+As far as PDO is concerned, CCF based TP is functionally identical to the Sawtooth based
+TP (except for one additional feature described below). A key difference beween the two ledgers
+is the fact that while the ledger in CCF is encrypted as noted above, the ledger is Sawtooth is stored in plain text.
+Even though the no part of conract state gets stored in the ledger in both CCF & Sawtooth, encrypting the
+ledger as in CCF helps to hide transaction patterns that are otherwise visible in Sawtooth ledger. Detailed documentation
+about Sawtooth based TP can be found at $PDO_SRC/sawtooth/docs. The schema for JSON payloads used to submit
+CCF transactions can be found at
+[${PDO_SRC}/python/pdo/submitter/ccf/docs/](../python/pdo/submitter/ccf/docs/ccf_payload_schema.json).
+For additional references to documentation about PDO, including transaction processor protocols,
+see [${PDO_SRC}/README.md](../README.md)
+
+A feature of the CCF based TP that is not supported by Sawtooth based TP is the fact that
```

Comment:
This paragraph here seems to be about API. I wonder if this can be re-organized as a section on new API calls, which then implicitly describes the underlying CCF features that the TP uses. But as above, I wouldn't really make any mention of Sawtooth here.

Reply:
thank you, please see answers above on documentation.